groups/bsl/bslstl/bslstl_optional.t.cpp

```diff
@@ -860,16 +860,23 @@ class TestDriver {
     // type.
   public:
     static void testCase2();
+    static void testCase2_Imp();
         // Optional_DataImp TEST.  The test requires TYPE to be default
         // constructible.
 
     static void testCase1();
         // Optional_Data TEST
 
 };
-
 template <class TYPE>
 void TestDriver<TYPE>::testCase2()
+{
+    testCase2_Imp();
+    TestDriver<const TYPE>::testCase2_Imp();
```

Comment:
Yeah, this asymmetry is unsavory (I missed the fact that this is a template class, not a template function).  I suggest a quick change:  make `testCase2_Imp` a free function template and have this function look more symmetric.

Reply:
As for the doc, I think it's good as is (i.e., detailed test description being in the `_Imp` function).

Reply:
yeah, I see what you mean. testCase2_Imp is now a free function template